### PR TITLE
feat(ui) build fetched images table

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Built FetchedImages table which is used to sync images with a different source

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/images and click "Change source"
- Use a source type that's different to the current selection and click "Connect"
- Check that an image selection table shows
- Select an image and click "Update selection"
- It might take a while but eventually you should get taken back to the main image list, and the header should say "Syncing images from {the source URL you chose}"
- Wait until images start downloading and check that the "Change source" button becomes disabled
- Click "Stop import".
- It might take a while but eventually downloading will stop, then check that the "Change source" button becomes enabled again.

## Fixes

Fixes canonical-web-and-design/app-squad#141

### CODE REVIEW

File: ui/src/app/images/components/UbuntuImageSelect/ArchSelect/ArchSelect.tsx [hunk 1]
```diff
@@ -4,14 +4,15 @@ import { useSelector } from "react-redux";
...
+  BaseImageFields,
   BootResourceUbuntuArch,
   BootResourceUbuntuRelease,
 } from "app/store/bootresource/types";
 import configSelectors from "app/store/config/selectors";
 
 type Props = {
   arches: BootResourceUbuntuArch[];
-  release: BootResourceUbuntuRelease | null;
+  release: BootResourceUbuntuRelease | BaseImageFields | null;
```
Comment (Caleb-Ellis) [inline ui/src/app/images/components/UbuntuImageSelect/ArchSelect/ArchSelect.tsx:15]: I've had to update a lot of the `UbuntuImageSelect` props to work with data coming from both `ubuntu.releases` and `fetchedImages.releases`. I think this is a bug in the backend as they should be the same, i.e both have the key `unsupported_arches`

Comment (huwshimi) [reply ui/src/app/images/components/UbuntuImageSelect/ArchSelect/ArchSelect.tsx:15]: Might be worth leaving a comment about when you get the different types.

**Review by huwshimi — APPROVED** [review]
This looks good. I think this is starting to take shape nicely.

One thing I found the first time I went through the flow is that when I got to the "Showing images fetched from..." it wasn't immediately obvious that I was still in the flow and not back at the main list. I wonder if the forms should be shown in the form card and then it's clear when you're still working through the flow (the machine list etc. use the card when you're shown the add forms).

One other thing was that when I pressed cancel from the second form (the "showing images..." screen) it went back to the "Choose source" form, but I think maybe it makes more sense to cancel the flow completely (otherwise you have to press cancel twice).

File: ui/src/app/images/components/UbuntuImageSelect/ArchSelect/ArchSelect.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/images/components/UbuntuImageSelect/ArchSelect/ArchSelect.tsx:15]: Yeah good call

**Comment by Caleb-Ellis** [comment]
> This looks good. I think this is starting to take shape nicely.
> 
> One thing I found the first time I went through the flow is that when I got to the "Showing images fetched from..." it wasn't immediately obvious that I was still in the flow and not back at the main list. I wonder if the forms should be shown in the form card and then it's clear when you're still working through the flow (the machine list etc. use the card when you're shown the add forms).
> 
> One other thing was that when I pressed cancel from the second form (the "showing images..." screen) it went back to the "Choose source" form, but I think maybe it makes more sense to cancel the flow completely (otherwise you have to press cancel twice).

I can fix the second comment now, but maybe the first comment can be part of the UX images rework issue? I remember Lyubo saying we're trying to move away from using cards for forms (or maybe it was just the highlighted cards). But yeah I can see how right now there's really not much difference between the main list and the fetched list.

**Comment by huwshimi** [comment]
> > This looks good. I think this is starting to take shape nicely.
> > One thing I found the first time I went through the flow is that when I got to the "Showing images fetched from..." it wasn't immediately obvious that I was still in the flow and not back at the main list. I wonder if the forms should be shown in the form card and then it's clear when you're still working through the flow (the machine list etc. use the card when you're shown the add forms).
> > One other thing was that when I pressed cancel from the second form (the "showing images..." screen) it went back to the "Choose source" form, but I think maybe it makes more sense to cancel the flow completely (otherwise you have to press cancel twice).
> 
> I can fix the second comment now, but maybe the first comment can be part of the UX images rework issue? I remember Lyubo saying we're trying to move away from using cards for forms (or maybe it was just the highlighted cards). But yeah I can see how right now there's really not much difference between the main list and the fetched list.

Sure.

**Pull request merged**